File: Herencia/ejercicio 1/Python/main.py
from Moto import Moto
from Coche import Coche
v2025 = [
    Coche("Toyota", "Corolla", 2025, 25000, 4, "Gasolina"),
    Coche("Ford", "Explorer", 2025, 35000, 5, "Diesel"),
    Moto("Yamaha", "R1", 2025, 18000, 1000, "RGB"),
    Moto("Honda", "CB500X", 2025, 12000, 500, "RGBXD"),
]
tc = []
tm = []
print("Vehiculos 2025")
for vehiculo in v2025:
    print(vehiculo.mostrar())
    if isinstance(vehiculo, Coche):
        tc.append(vehiculo)
    elif isinstance(vehiculo, Moto):
        tm.append(vehiculo)
print("tiene 4 puertas")
for coche in tc:
    if coche.nPuertas > 4:
        print(coche.mostrar())

File: Herencia/ejercicio 1/Java/main.java
import java.util.ArrayList;
public class main {
    public static void main(String[] args) {
        Vehiculo[] v2025 = new Vehiculo[4];
        v2025[0] = new Coche("Ford", "Focus", 2025, 20000, 4, "Diesel");
        v2025[1] = new Coche("Ford", "Focus", 2025, 20000, 4, "Diesel");
        v2025[2] = new Moto("Yamaha", "R1", 2025, 15000, 1000, "Gasolina");
        v2025[3] = new Moto("Yamaha", "R1", 2025, 15000, 1000, "Gasolina");
        ArrayList<Coche> coches = new ArrayList<>();
        ArrayList<Moto> motos = new ArrayList<>();
        System.out.println("Vehiculos 2025");
        for (Vehiculo vehiculo : v2025) {
            if (vehiculo instanceof Coche) {
                coches.add((Coche) vehiculo);
            } else if (vehiculo instanceof Moto) {
                motos.add((Moto) vehiculo);
            }
        }
        System.out.println("Coches con mas de 4 puertas");
        for (Coche coche : coches) {
            System.out.println(coche.mostrar());
        }
    }   
}
